.github/workflows/haskell-ci.yml

```diff
@@ -72,6 +72,20 @@ on:
         # For an explanation of `--ghc-options='-j'` and `--semaphore`, see
         # Note [Parallelism].
         default: --ghc-options='-j' --semaphore --enable-tests --minimize-conflict-set
+      coverage:
+        description: |
+          Whether to run `hpc report` to report test-suite coverage.
+
+          Note that the client must enable some form of coverage tracking, e.g.,
+          by adding `--enable-coverage` to the `configure-flags` input.
+        type: boolean
+        required: false
+        default: true
```

Comment:
Enabling coverage by default could be considered a pitfall, given that you need to enable coverage at the `cabal` level separately in order for the coverage step to succeed. I am aware that the coverage step is allowed to fail, but the status quo means that any use of `haskell-ci` by default will result in an extra, failing step unless you take additional action to prevent this.